main.cpp

```diff
@@ -115,17 +140,19 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T t1( 9, "Nine" );                                             
+    T t2( 7, "Seven" );                                             
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    Comparison f;                                            
+    auto* smaller = f.compare( &t1 , &t2 );                              
     
-    U <#name3#>;
-    float updatedValue = 5.f;
-    std::cout << "[static func] <#name3#>'s multiplied values: " << <#structname2#>::<#staticFunctionA#>( , ) << std::endl;                  //11
+    if( smaller )
+        std::cout << "the smaller one is << " << smaller->name << std::endl;
```

Comment:
Be explicit when checking pointes and whether or not they `== nullptr`
while this is concise, the intent regarding what you're checking is not as obvious as it could be.
- [ ] revise